### ISSUE TITLE
Using `keypress` for shortcuts with modifiers make them unusable with Dvorak

### ISSUE THREAD

**Issue by mikaeldui** [openstreetmap/iD#9889]
### How to reproduce the issue?

1. Switch to Dvorak
2. Try simple commands like Ctrl+C, C, M, or anything else
3. Note that the shortcuts have been bound to the letter on the key instead of the physical key

This makes them practically impossible to use. The solution is to use `keydown` instead of `keypress`, so that Ctrl+C is where C is on the QWERTY keyboard, and not Ctrl+I as on Dvorak.

```
document.addEventListener("keydown", e => console.log("keydown: " + e.which));
document.addEventListener("keypress", e => console.log("keypress: " + e.which));
```

### Which deployed environments do you see the issue in?

Released version at openstreetmap.org/edit

### Which browsers are you seeing this problem on?

Microsoft Edge

**Comment by pnorman** [comment]
Is the best practice for shortcuts on Dvorak to bind them to location instead of letter? Does this present any complications with keyboard layouts from other languages?

If this is done, the documentation would also need updating.

**Comment by mikaeldui** [comment]
> Is the best practice for shortcuts on Dvorak to bind them to location instead of letter? Does this present any complications with keyboard layouts from other languages?
> 
> If this is done, the documentation would also need updating.

All of the Windows apps use location, but I've noticed some web apps that don't. So, one can select text and copy/paste with the location, but I can't copy/paste iD objects using the same key locations.

**Comment by 1ec5** [comment]
The accelerators (the shortcuts without modifiers) are localizable by language, consistent with Windows and Mac conventions. Not every language has localized the modifiers, but some have. It can be pretty difficult to choose localized accelerator keys that remain ergonomic, but in my opinion, using memorable mnemonics is more important than consistency with the fingers that other languages’ speakers press. Binding the accelerators to QWERTY locations might work for English and some other languages like Japanese that favor QWERTY in some contexts, but it would get messier in other languages.

**Comment by 1ec5** [comment]
As for the keyboard combinations (the ones with modifiers), I agree that platform conventions would call for keeping the locations constant across localizations and keyboard layouts.

**Comment by mikaeldui** [comment]
> As for the keyboard combinations (the ones with modifiers), I agree that platform conventions would call for keeping the locations constant across localizations and keyboard layouts.

I agree that the shortcuts without modifiers (some with modifiers: Ctrl+Z, Ctrl+Y, Ctrl+X, Ctrl+C and Ctrl+V) could use `keypress`.

**Comment by mikaeldui** [comment]
@GitHub has the same bug/feature and their support replied with this:

> Typing ? on GitHub brings up a dialog box that lists the keyboard shortcuts available for that page. I may have misunderstood but I take it you're pressing the keys in the position of what Ctrl+C would be on a QWERTY keyboard, not the keys that are Ctrl+C in your layout?
> This is a common hurdle faced by users of international or custom keyboard layouts and there's no simple solution. Often users make use of scripts or software to temporarily switch layouts when holding the control key, or remap command keys entirely to workaround it.
> Our keyboard shortcuts rely on discoverability through our documentation - in turn this documentation relies on rendering the printable character, and so in turn we must rely on what would be the printed character, not the keycode.
> If you prefer, you can disable character key shortcuts from your [accessibility settings](https://github.com/settings/accessibility). For more information, read [Managing keyboard shortcuts](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/managing-accessibility-settings#managing-keyboard-shortcuts) in the GitHub Docs.
> Please let me know if you need anything else.
